internal/version/release_version.go

```diff
@@ -40,35 +39,30 @@ type releaseVersionFetcher struct {
 	ctx context.Context
 }
 
-// NVersions retrieves the first n versions returned by Github list releases endpoint.
-func (s *releaseVersionFetcher) NVersions(n int) ([]*github.RepositoryRelease, error) {
-	var releases []*github.RepositoryRelease
+// LatestWithCriteria retrieves the first release version that matches the criteria. We assume that ListReleases returns releases sorted by created_at value.
+func (s *releaseVersionFetcher) LatestWithCriteria(n int, matchCriteria func(tag string, tool string) bool, toolName string) (*github.RepositoryRelease, error) {
```

Comment:
nit: I wouldn't expose a matcher parameter here because I don't think we would ever change the behaviour at runtime

Reply:
Trying to keep this logic in `tool_latest_version_finder` and when comparing the times it doesn't take much